src/library/commissioner_safe.cpp

```diff
@@ -96,9 +95,9 @@ Error CommissionerSafe::Start()
 {
     Error error = Error::kNone;
 
-    VerifyOrExit(mEventThread == nullptr, error = Error::kAlready);
+    VerifyOrExit(!mEventThread.joinable(), error = Error::kAlready);
```

Comment:
Could two threads calls `Start()` at the same time and `mEventThread.joinable()` returns False for both thread. Both threads will then try to create `mEventThread`?

I suspect all methods of `CommissionerSafe` need to be thread safe?

Reply:
Yes it is not thread safe. I think the thread safety promise of this class is much. I updated the [doc string of class CommissionerSafe](). Please check if it makes sens for you.

Reply:
Sure, let's keep it simple.